cranelift/filetests/src/function_runner.rs

```diff
@@ -611,6 +611,7 @@ extern "C" fn __cranelift_x86_pshufb(a: __m128i, b: __m128i) -> __m128i {
 }
 
 #[cfg(test)]
+#[cfg(target_pointer_width = "64")] // 32-bit platforms not supported at this time
```

Comment:
The non-support here has less to do with "32-bit" than "no Cranelift backend for the host architecture", right? I wonder if we could add some build-script logic to add a `cranelift-has-host-backend` feature and then `cfg` on that here? My concern is otherwise that a bunch of "exclude 32-bit" conditions scattered through tests get missed if/when we do add a native backend for say arm32 or riscv32.

Reply:
I'll try to think of something better yeah, but I also don't want to go too too overboard here because the tests handled in this PR are about 1% of our tests and are probably all largely redundant with all other tests we do from a "run this on all platforms" perspective. I definitely agree with what you're thinking though, I'm also not a fan of writing these #[cfg] and I like the idea of a `build.rs`-based thing (although it may be difficult to propagate that out beyond cranelift-codegen...)

Reply:
Alternate idea: could we add a method to `Engine` (or `Config` or ...) that lets the user query whether Wasmtime has a native compilation backend? Then the affected tests could early-out succeed. Such a query is probably useful for other cases too...

Reply:
(Likewise at the `cranelift-native` level, or whatever other level, for Cranelift.)

Reply:
It might help yeah, but the problem is that we have tests all over the place in all sorts of crates that need this logic. That leads to either duplication of the logic, moving tests, or something like that. Basically it's just really difficult wrangling a ubiquitous dependency on platform-specific logic when testing across platforms.

Reply:
I guess we could push and dedupe the check into some common dependency, which is a bit annoying, but would otherwise work. `wasmtime-environ`?

There is someone actively working on a riscv32 backend for cranelift, they've been commenting on zulip, so it would be a shame to build in new cranelift-doesn't-support-32-bit assumptions at this point...